crates/rmf_site_camera/src/systems.rs

```diff
@@ -156,42 +156,66 @@ pub fn init_cameras(
 pub fn change_projection_mode(
     projection_mode: Res<ProjectionMode>,
     mut cameras: Query<(&mut Camera, &mut Visibility)>,
-    ortho_cams: Query<Entity, With<OrthographicCameraRoot>>,
-    persp_cams: Query<Entity, With<PerspectiveCameraRoot>>,
+    ortho_cams: Query<(Entity, &Children), With<OrthographicCameraRoot>>,
+    persp_cams: Query<(Entity, &Children), With<PerspectiveCameraRoot>>,
```

Comment:
Why not just combine the `cameras` query component into the `ortho_cams` or `persp_cams`? This would also get rid of some of the `let else` statements below.
I suppose doing the naive way would panic because of conflicting access for mutable variables but you could use a disjoint query such as:

`ortho_cams: Query<(Entity, &Children, &mut Camera, &mut Visibility), (With<OrthographicCameraRoot>, Without<PerspectiveCameraRoot>)>`

And viceversa for the perspective cameras

Reply:
I might be understanding you wrongly, but `OrthographicCameraRoot` and `PerspectiveCameraRoot` are only inserted in the base cameras, so we wouldn't be able to query for the `Camera` of their children.

Reply:
There are several ways we could consider tweaking this implementation, like we could introduce `OrthographicUserCamera` and `PerspectiveUserCamera` components which we apply to to their respective camera entities, and then we can have a query that filters on those components instead of iterating through children.

But I think this fix is important enough that I don't want to spend longer than necessary deliberating. We can always follow up with a more elegant revision later if we decide to do a cleanup of the codebase.